flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobServerTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.blob;
+
+import org.apache.flink.configuration.BlobServerOptions;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.util.OperatingSystem;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import javax.annotation.Nonnull;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeTrue;
+
+/**
+ * Tests for the {@link BlobServer}.
+ */
+public class BlobServerTest extends TestLogger {
+
+	@ClassRule
+	public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+	/**
+	 * Tests that the {@link BlobServer} fails if the blob storage directory
+	 * cannot be created.
+	 */
+	@Test
+	public void testFailureIfStorageDirectoryCannotBeCreated() throws IOException {
+		assumeTrue(!OperatingSystem.isWindows()); //setWritable doesn't work on Windows.
```

Comment:
I would prefer to move the _assumption_ closer to `createNonWritableDirectory`. This makes the comment clearer. Also consider `assumeFalse` as `!` can be overlooked.
```suggestion
		assumeFalse(OperatingSystem.isWindows()); //setWritable doesn't work on Windows.
		final File blobStorageDirectory = createNonWritableDirectory();

		final Configuration configuration = new Configuration();
		final String nonExistDirectory = new File(blobStorageDirectory, "does_not_exist_for_sure").getAbsolutePath();
		configuration.setString(BlobServerOptions.STORAGE_DIRECTORY, nonExistDirectory);
```

Reply:
Good point Gary. I will move the assumption into the `createNonWritableDirectory` method and change it to `assumeFalse`.